elastic/fleet-server
# update go mod elatic/beats reference with newer commit

## What is the problem this PR solves?

Update the version of elastic/beats to from `v7.11.1` to a more recent one. Specifically https://github.com/elastic/beats/commit/4641f759226c1fbf00e0ec1c9012538e5936279a off the `8.0` branch.

## How does this PR solve the problem?

Updated module includes libbeat support `ssl.ca_trusted_fingerprint` (https://github.com/elastic/beats/pull/29229). This option is used by libbeat to trust a self-signed CA when bootstrapping a secure by default cluster in 8.0.
The fleet-server will use this option when connecting to ES.

## How to test this PR locally

The elastic-agent modifications to support/pass this option to fleet-server can be found with: https://github.com/elastic/beats/pull/29128.

Create a snapshot build of the agent, and fleet-server.

start an 8.0+ ES node and Kibana
go to Kibana and setup fleet (ensure that the elasticsearch scheme is `https`.
enroll the fleet server with the command and the `--fleet-server-es-ca-trusted-fingerprint` flag, using the `sha256` fingerprint that elasticsearch outputs on startup. i.e.:

```sh
./elastic-agent enroll  \
  --fleet-server-es=https://localhost:9200 \
  --fleet-server-service-token=AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL3Rva2VuLTE2Mzg5MDYzMTgyNTA6TG1NZjJlMXNUQWVYS3FEbW95bHFBZw \
  --fleet-server-policy=0dd498f0-578b-11ec-89c7-8f0ea2875250 \
  --fleet-server-es-ca-trusted-fingerprint=ab12381f29e755c657ba451df9505d27317cf295062ae949ae2eb10059f31875
```

## Related issues

- Relates #760
